### DISCUSSION TITLE
Watervapor in unsaturated soil (PorousFlow)

### DISCUSSION THREAD

**Discussion by marinsiebert** [idaholab/moose#23541] [Q&A Modules: General] [answered]
Hello everyone, 

I have stumbled into an issue which I am not sure how to resolve. I am trying to model a partially saturated soil (S < 1) with the PorousFlow module. the PorousFlowCapillaryPressureVG is used to calculate the saturations, which is currently done by the PorousFlow1PhaseP with negative porepressures according to Pp = - Pc. 
However, it is not possible to apply the Water97FluidProperties in this case, as this object requires pressures > 0. 
How would one approach this issue? The goal would be to investigate evaporisation processes in partially saturated soils due to heat source.

**Comment by GiudGiud** [comment]
@Rliu2020

**Comment by marinsiebert** [comment]
As the issue is still unresolved kindly giving this a push, in the hopes of receiving some info.

**Reply by cpgr** [reply]
@WilkAndy - any ideas?

**Comment by WilkAndy** [comment, marked as answer]
I have been decoupled from MOOSE for the last month, I'm sorry.  In single-phase systems, the convention is to set porepressure = 0 at the transition between full saturation and saturation < 1.  PorousFlow uses this convention.  However, in reality we know that at this point, porepressure equals atmospheric pressure (eg, 101kPa).   Therefore, if you are going to use Water97FluidProperties or any other FluidProperties equation of state that uses true pressure (and not the single-phase "i'm going to pretend there is no atmospheric pressure by defining porepressure = 0 at the transition to full saturation") then you have to add atmospheric pressure on to PorousFlow's single-phase porepressure.

Eg, if PorousFlow's single-phase porepressure = 0, then you've got to evaluate the FluidProperties using pressure = 101kPa.
Eg if PorousFlow's single-phase porepressure = 8MPa, then you've got to evaluate the FluidProperties using pressure = 8101kPa
Eg if PorousFlow's single-phase porepressure = -50kPa, then you've got to evaluate the FluidProperties using pressure = 51kPa

This is really annoying, I admit, but is just the convention used by single-phase solvers.  The problem really comes when the PorousFlow's single-phase porepressure = -1MPa (or something < -atmospheric_pressure), because then you'll end up supplying FluidProperties with a negative pressure, and your simulation will crash.

So....   my recommendation is to use TabulatedFluidProperties: https://mooseframework.inl.gov/moose/source/userobjects/TabulatedFluidProperties.html and manually do the shift with atmospheric pressure, and also (critically) extend to negative pressure.  This extension will likely be very important for convergence, even if your physical system never enters this regime in reality, because MOOSE will explore the regime as it iterates towards the solution.  I suggest do NOT simply use constant density and viscosity in this regime, but instead use some sort of exponentially-decaying thing as porepressure goes to negative infinity.  Eg, use a constant bulk modulus.  The problem with constant density (and viscosity) is that, because density doesn't change, MOOSE sees no penalty in making porepressure very negative - it might be essentially solving  Ax = 0, and with A = 0, then x can be anything.

A

**Reply by WilkAndy** [reply]
Sorry, i should have tagged @marinsiebert

**Reply by WilkAndy** [reply]
I also see there are two open issues on this: https://github.com/idaholab/moose/issues/21794 and https://github.com/idaholab/moose/issues/22740

**Reply by marinsiebert** [reply]
Thank you for the detailed answer, it is much appreciated. I avoided the issue for now by switching to the 2PhasePS formulation, but will now go back and try to make this work.
